### PR TITLE
Update SIP-007

### PR DESCRIPTION
**Pull request by kantai**
New features:

* Threshold adjustment based on participation-level
* Time-windowing PoX
* Discussion of Stacks and Bitcoin broadcast of signed Stacker messages.

### CODE REVIEW

File: sip/sip-007-stacking-consensus.md [hunk 1]
```diff
@@ -211,6 +212,62 @@ cycles are fixed length --- if fewer addresses participate in the
...
+  STX participate in a reward cycle, participant wallets controlling `x`
+  STX may include `floor(x / (0.00010*STX_LIQUID_SUPPLY))` addresses in
+  the reward set. (i.e., the minimum participation threshold is 1/10,000th
+  of the liquid supply).
+* **75%** If between `0.5 * STX_LIQUID_SUPPLY` and `0.75 * STX_LIQUID_SUPPLY`
+  STX participate in a reward cycle, participant wallets controlling `x`
+  STX may include `floor(x / (0.00015*STX_LIQUID_SUPPLY))` addresses in
+  the reward set. (i.e., the minimum participation threshold is 3/20,000th
+  of the liquid supply).
+* **75%** If more than `0.75 * STX_LIQUID_SUPPLY` STX participate in a
```
Comment (jcnelson) [inline sip/sip-007-stacking-consensus.md]: Should this be `100%`?

File: sip/sip-007-stacking-consensus.md [hunk 2]
```diff
@@ -211,6 +212,62 @@ cycles are fixed length --- if fewer addresses participate in the
...
+
+1. Indicating to the network how many STX should be locked up, and for
+   how many reward cycles.
+2. Indicate support for a particular chain tip.
+3. Specifying the Bitcoin address for receiving Stacking rewards.
+
+These messages may be broadcast either on the Stacks chain or the
+Bitcoin chain. If broadcast on the Stacks chain, these messages must
+be included in the Stacks chain _before_ the anchor block for the
+reward period. If broadcast on the Bitcoin chain, they may be
```
Comment (jcnelson) [inline sip/sip-007-stacking-consensus.md]: Recommend adding something like: "Moreover, if sent to the Stacks chain, the anchor block must _confirm_ the transaction bearing the signed message."

I don't think this is true if the transaction is sent on the Bitcoin chain.  Sending a transaction on the Bitcoin chain effectively registers the chain tip signal in all subsequently-mined Stacks forks on that Bitcoin fork.

Comment (jcnelson) [reply sip/sip-007-stacking-consensus.md]: The sentence "If broadcast on the Stacks chain, these messages must ￼be included in the Stacks chain _before_ the anchor block for the ￼reward period" is confusing to me.  Does it mean that if I signal on the Stacks chain, I have to get my transaction in before the prepare phase for the next reward cycle starts?  i.e. I have to get it in during the previous reward cycle, in order to be included in the next reward cycle?

Comment (kantai) [reply sip/sip-007-stacking-consensus.md]: Yes -- the anchor block is the _latest possible_ view of Stacks data for the current reward cycle. That means that a stacking-message on the Stacks chain would need to be confirmed before the prepare phase begins.

Comment (kantai) [reply sip/sip-007-stacking-consensus.md]: > I don't think this is true if the transaction is sent on the Bitcoin chain. Sending a transaction on the Bitcoin chain effectively registers the chain tip signal in all subsequently-mined Stacks forks on that Bitcoin fork.

True, but the message still "picks" a chain tip. If a message signals support for a chain tip that ends up not being the anchor block, should they qualify for Stacking rewards?

Comment (jcnelson) [reply sip/sip-007-stacking-consensus.md]: No, they shouldn't receive rewards in that case.  But, just based on the text here, I don't know whether or not it's the case that the anchor block confirms any Bitcoin-based signaling transactions.

**Comment by jcnelson** [comment]
Thanks for the update @kantai!  LGTM

**Comment by kantai** [comment]
> Why are we proposing to shrink the number of levels?

The idea is to remove the discretization effects from the threshold levels. Reward magnification risks are highest at low participation rates, so we set a limit there, but otherwise, the discretization isn't providing much benefit, and leads to strange outcomes at the borders.

**Pull request merged**